File: python/0881-boats-to-save-people.py
class Solution(object):
    def numRescueBoats(self, people: list[int], limit: int) -> int:
        people.sort()
        right = len(people) - 1
        left = res = 0
        while left <= right:
            if people[left] + people[right] <= limit:
                left += 1
            right -= 1
            res += 1
        return res

File: kotlin/0881-boats-to-save-people.kt
class Solution {
    fun numRescueBoats(people: IntArray, limit: Int): Int {
        people.sort()

        var left = 0
        var right = people.lastIndex
        var res = 0

        while (left <= right) {
            if (people[left] + people[right] <= limit)
                left++
            right--
            res++
        }

        return res
    }
}
